pkg/controller/servicebinding/secret_test.go

```diff
@@ -38,17 +37,100 @@ func TestSecretNew(t *testing.T) {
 		f.FakeDynClient(),
 		ns,
 		name,
+		data,
+		secretOwnerReference,
 	)
 
-	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName, err := HashedResourceNameForData(name, data)
 		assert.NoError(t, err)
+		assert.Contains(t, hashedName, name)
+
+		hashedName2, err := HashedResourceNameForData(hashedName, data)
```

Comment:
This use case feels that `sbr.Status.Secret` is being used in place of `sbr.GetName()` that shouldn't; I'd urge to fix the reason of this there as well.

Regarding this use case, I find it un-expected.

Reply:
This test `hashedName2, err := HashedResourceNameForData(hashedName, data)` is there to show that datahash is being appended only once. 

There have been scenarios when dataHash was appended twice like this -> `sb-1-72ddc0c540ab3a290e138726940591debf14c581-72ddc0c540ab3a290e138726940591debf14c581`

Reply:
So, if the secret name already contains the calculated hash.. the datahash is not appended again.

Reply:
> This use case feels that sbr.Status.Secret is being used 

It feels like that but it's not.
There is no ServiceBinding here.

I am not assigning the value `hashedName` to any ServiceBinding's status.

I am just testing the function `HashedResourceNameForData` by providing it a string which already contains the data hash.

Reply:
> > This use case feels that sbr.Status.Secret is being used
> 
> It feels like that but it's not.
> There is no ServiceBinding here.
> 
> I am not assigning the value `hashedName` to any ServiceBinding's status.
> 
> I am just testing the function `HashedResourceNameForData` by providing it a string which already contains the data hash.

I understand what this test does. 

I'm pointing out that, the reason for this specific use case to exist is due to a defect somewhere else and that defect should be fixed, and that is likely to be a code-path where `sbr.Status.Secret` is used as name of the secret and it shouldn't, in this very same changelist.

Reply:
> I'm pointing out that, the reason for this specific use case to exist is due to a defect somewhere else and that defect should be fixed

Oh Okay. 

> that is likely to be a code-path where sbr.Status.Secret is used as name of the secret and it shouldn't

Let me look

Reply:
Cannot reproduce that anymore. Removed that.